sw/vendor/riscv_compliance/riscv-target/opentitan/device/rv32imc/link.ld

```diff
@@ -0,0 +1,55 @@
+/* Copyright lowRISC contributors.
+   Licensed under the Apache License, Version 2.0, see LICENSE for details.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+OUTPUT_ARCH( "riscv" )
+ENTRY(_start)
+
+/* required to correctly link newlib */
+GROUP( -lc -lgloss -lgcc -lsupc++ )
+
+SEARCH_DIR(.)
+__DYNAMIC  =  0;
+
+MEMORY
+{
+    flash (rx)     : ORIGIN = 0x20000000, LENGTH = 0x100000
+    ram (!rx)      : ORIGIN = 0x10000000, LENGTH = 0x10000
```

Comment:
These attributes seem strange.  It will place sections that are neither read or execute into the ram region.  Shouldn't it just be `(rwx)`?

Reply:
hmmmm i followed the examples here


I think r is read-only.  So !rx to me is interpreted as not-read-only, executable.  I guess I could do (wx) also? W is read/write apparently.

Reply:
AIUI anything past the `!` [is negated](https://sourceware.org/binutils/docs/ld/MEMORY.html).  So `(!rx)` would mean anything that is _not_ read and _not_ execute will go into `ram`. I do think it's moot, because the section definitions below manually place things into flash/ram, but I think `(wx)` is more correct.

Reply:
yeah i'll change to wx